runtime/gc_structs/ClassLoaderClassesIterator.cpp

```diff
@@ -99,6 +102,57 @@ GC_ClassLoaderClassesIterator::switchToSystemMode()
 	return isSystemClassLoader;
 }
 
+J9Class *
+GC_ClassLoaderClassesIterator::nextArrayClass()
+{
+	while (_arrayState != state_done) {
+		switch(_arrayState) {
+			case state_valueTypeArray:
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+				if (NULL != J9CLASS_GET_NULLRESTRICTED_ARRAY(_startingClass)) {
+					_iterateArrayClazz = J9CLASS_GET_NULLRESTRICTED_ARRAY(_startingClass);
+					_arrayState = state_valueTypeArrayList;
+				} else
+#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
+				if (NULL != _startingClass->arrayClass) {
+					_arrayState = state_array;
+				} else {
+					_arrayState = state_done;
+				}
```

Comment:
You can replace 
```
    else
...
    if (NULL != _startingClass->arrayClass) {
        _arrayState = state_array;
    } else {
        _arrayState = state_done;
    }
```
to
```
    else
...
    {
        _arrayState = state_array;
    }
```
I'm ok with this as a shortcut. Checks are the same, just with one extra state and one while loop iteration

Reply:
Updated.